src/messages/Image/Image.tsx

```diff
@@ -1,20 +1,22 @@
 import { FC, useMemo, useState } from "react";
-import { MessagePasstroughProps } from "../types";
 import { ImageMessageContext } from "./context";
 import Lightbox from "./lightbox/Lightbox";
 import ImageThumb from "./ImageThumb";
 import { useMessageContext } from "src/hooks";
+import { getChannelPayload } from "src/utils";
+import { IWebchatButton } from "@cognigy/socket-client/lib/interfaces/messageData";
 
-const Image: FC<MessagePasstroughProps> = () => {
-	const { message } = useMessageContext();
+const Image: FC = () => {
+	const { message, config } = useMessageContext();
+	const payload = getChannelPayload(message, config);
+	const { url, altText, buttons } = payload.message.attachment.payload;
```

Comment:
maybe need a check here if payload.message.attachment.payload is valid?

Reply:
We are already checking in src/matcher.ts 
if you reach the component level at least the attachment property is defined. It could be possible that, from endpoint data, attachment is defined and attachment.payload not? In that case I would add a safe check on that last property.

Reply:
fixed